### PR TITLE
consensus epoch number fix

### PR DESCRIPTION
**Pull request by raduchis**
computation of the consensus group for the start of epoch block was computed with the current epoch not the previous epoch

- fixed consensus group computation for start of epoch with previous epoch both on shards and meta
- added unit tests for UpdateState method

### CODE REVIEW

File: process/mock/peerAccountHandlerMock.go [hunk 1]
```diff
@@ -95,28 +100,33 @@ func (p *PeerAccountHandlerMock) GetShardId() uint32 {
...
 	if p.DecreaseLeaderSuccessRateCalled != nil {
 		p.DecreaseLeaderSuccessRateCalled(val)
 	}
+	p.DecreaseLeaderSuccessRateValue += val
 }
 
 // IncreaseValidatorSuccessRate -
 func (p *PeerAccountHandlerMock) IncreaseValidatorSuccessRate(val uint32) {
 	if p.IncreaseValidatorSuccessRateCalled != nil {
 		p.IncreaseValidatorSuccessRateCalled(val)
```
Comment (SebastianMarian) [inline process/mock/peerAccountHandlerMock.go:119]: Should be added return ?

File: process/mock/peerAccountHandlerMock.go [hunk 2]
```diff
@@ -95,28 +100,33 @@ func (p *PeerAccountHandlerMock) GetShardId() uint32 {
...
 		p.IncreaseLeaderSuccessRateCalled(val)
+		return
 	}
+	p.IncreaseLeaderSuccessRateValue += val
 }
 
 // DecreaseLeaderSuccessRate -
 func (p *PeerAccountHandlerMock) DecreaseLeaderSuccessRate(val uint32) {
 	if p.DecreaseLeaderSuccessRateCalled != nil {
 		p.DecreaseLeaderSuccessRateCalled(val)
```
Comment (SebastianMarian) [inline process/mock/peerAccountHandlerMock.go:111]: Should be added return ?

File: process/mock/peerAccountHandlerMock.go [hunk 3]
```diff
@@ -95,28 +100,33 @@ func (p *PeerAccountHandlerMock) GetShardId() uint32 {
...
 	if p.IncreaseValidatorSuccessRateCalled != nil {
 		p.IncreaseValidatorSuccessRateCalled(val)
 	}
+	p.IncreaseValidatorSuccessRateValue += val
 }
 
 // DecreaseValidatorSuccessRate -
 func (p *PeerAccountHandlerMock) DecreaseValidatorSuccessRate(val uint32) {
 	if p.DecreaseValidatorSuccessRateCalled != nil {
 		p.DecreaseValidatorSuccessRateCalled(val)
```
Comment (SebastianMarian) [inline process/mock/peerAccountHandlerMock.go:127]: Should be added return ?

File: process/peer/process.go [hunk 4]
```diff
@@ -354,6 +355,17 @@ func (vs *validatorStatistics) UpdatePeerState(header data.HeaderHandler, cache
 	return rootHash, nil
 }
 
+func computeEpoch(header data.HeaderHandler) uint32 {
+	// TODO: change if start of epoch block needs to be validated by the new epoch nodes
+	// previous block was proposed by the consensus group of the previous epoch
+	epoch := header.GetEpoch()
+	if (header.IsStartOfEpochBlock()) && epoch > 0 {
```
Comment (SebastianMarian) [inline process/peer/process.go:362]: No need for open-close ()

File: process/peer/process.go [hunk 5]
```diff
@@ -734,17 +745,22 @@ func (vs *validatorStatistics) updateShardDataPeerState(
...
 	for _, h := range metaHeader.ShardInfo {
 		if h.Nonce == vs.genesisNonce {
 			continue
 		}
 
+		currentHeader, ok := cacheMap[string(h.HeaderHash)]
+		if !ok {
+			return fmt.Errorf("%w - updateShardDataPeerState header from cache - hash: %s, round: %v, nonce: %v",
+				process.ErrMissingHeader,
+				hex.EncodeToString(h.GetPrevHash()),
```
Comment (SebastianMarian) [inline process/peer/process.go:757]: h.HeaderHash ?

File: process/peer/process_test.go [hunk 6]
```diff
@@ -596,6 +598,535 @@ func TestValidatorStatisticsProcessor_UpdatePeerStateCallsIncrease(t *testing.T)
 	assert.True(t, increaseValidatorCalled)
 }
 
+func TestValidatorStatisticsProcessor_UpdatePeerState_IncreasesConsensusPreviousMetaBlock_SameEpoch(t *testing.T) {
+	t.Parallel()
```
Comment (SebastianMarian) [inline process/peer/process_test.go:602]: Add empty line after each t.Parallel() in this file

Comment (raduchis) [reply process/peer/process_test.go:602]: fixed

File: process/peer/process.go (hunk 5 above)
Comment (raduchis) [reply process/peer/process.go:757]: fixed

File: process/peer/process.go (hunk 4 above)
Comment (raduchis) [reply process/peer/process.go:362]: fixed

File: process/mock/peerAccountHandlerMock.go (hunk 1 above)
Comment (raduchis) [reply process/mock/peerAccountHandlerMock.go:119]: yes, no need for the += afterwards

File: process/mock/peerAccountHandlerMock.go (hunk 2 above)
Comment (raduchis) [reply process/mock/peerAccountHandlerMock.go:111]: fixed

File: process/mock/peerAccountHandlerMock.go (hunk 3 above)
Comment (raduchis) [reply process/mock/peerAccountHandlerMock.go:127]: fixed

**Pull request merged**